### DISCUSSION TITLE
Default Secure Client-Initiated Renegotiation Vulnerability?

### DISCUSSION THREAD

**Discussion by 3XC1T3D** [keycloak/keycloak#25209] [Q&A] [answered]
Hi,

we are testing our SSL Configuration with https://testssl.sh/.

When we are running the test against our keycloak Installation (on Port 8443), it recognizes a Secure Client-Initiated Renegotiation Vulnerability.

`Secure Client-Initiated Renegotiation     VULNERABLE (NOT ok), DoS threat (6 attempts)`

I made some researches and found out, that it is possible to reject the Client-Initiated Renegotiation with following JVM Parameter:

` -Djdk.tls.rejectClientInitiatedRenegotiation=true`

Now i am wondering , why isnt this Parameter default in the keycloak installation/start process. Can anyone explain this to me?

best regards
Nico

**Comment by mikraecha** [comment]
Hi,

same finding:

```
Finding Information
Port:	8443 TCP - http
Name:	SSL/TLS Secure Client-Initiated Renegotiation Denial of Service Vulnerability
Risk Score:	5.0 (Medium risk)
Product:	SSL/TLS
Vulnerability Information	The client can request a renegotiation, which is more CPU intense for the server then for the client.
Solution:	Disable the secure client-Initiated renegotiation in the tls service.
```

Best,
Michael

**Comment by ahus1** [comment, marked as answer]
Issue raised here: https://github.com/keycloak/keycloak/issues/43020
